AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+/**
+ This is the snapshot shot image node, this will be hidden (but not nil) when .liveMap = YES
+ */
+@property (nonatomic, readonly) ASImageNode *mapImage;
+/**
+ This is the ASDisplayNode that backs the MKMapView. This will be nil if .liveMap = NO. To access the underlying MKMapView, in order to set a delegate for example, use (MKMapView *)mapView.view;
+ */
+@property (nonatomic, readonly) ASDisplayNode *mapView;
+/**
+ Set this to YES to turn the snapshot into an interactive MKMapView and vice versa. Defaults to NO.
+ */
+@property (nonatomic, assign, getter=isLiveMap) BOOL liveMap;
+/**
+ @abstract Explicitly set the size of the map and therefore the size of ASMapNode. Defaults to CGSizeMake(constrainedSize.max.width, 256).
+ @discussion If the mapSize width or height is greater than the available space, then ASMapNode will take the maximum space available.
+ @result The current size of the ASMapNode.
+ */
+@property (nonatomic, assign) CGSize mapSize;
+/**
+ @abstract Whether ASMapNode should automatically request a new map snapshot to correspond to the new node size. Defaults to YES.
+ @discussion If mapSize is set then this will be set to NO, since the size will be the same in all orientations.
+ */
+@property (nonatomic, assign) BOOL automaticallyReloadsMapImageOnOrientationChange;
+/**
+ Set the delegate of the MKMapView.
+ */
+@property (nonatomic, weak) id <MKMapViewDelegate> mapDelegate;
```

Comment:
Aha, this is nice to have.  Extend the comment to explain that this can be set even before the mapView is created, but will be set on the map in the case that the liveMap mode is engaged.